### PR TITLE
feat: don't restrict docs of formats in DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS

### PR DESCRIPTION
**Pull request by patte**
- [x] set ENV: `DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS=republik/format-covid-19-uhr-newsletter`

### CODE REVIEW

File: packages/search/graphql/resolvers/_queries/search.js [hunk 1]
```diff
@@ -329,9 +330,20 @@ const getFirst = (first, filter, user, recursive, unrestricted) => {
...
-  if (DOCUMENTS_RESTRICT_TO_ROLES && !recursive && !path && !oneRepoId && !unrestricted) {
+
+  const format = getFilterValue(filter, 'format')
+  const whitelisted = format && format.length && DOCUMENTS_UNRESTRICTED_REPO_IDS && DOCUMENTS_UNRESTRICTED_REPO_IDS
+    .split(',')
+    .some(repoId =>
+      format.some(f =>
+        new RegExp(`.*${repoId}$`).test(f)
+      )
+    )
```
Comment (patrickvenetz) [inline packages/search/graphql/resolvers/_queries/search.js]:
```suggestion
    .some(repoId => format.some(new RegExp(`.*${repoId}$`).test)
```

File: packages/search/graphql/resolvers/_queries/search.js [hunk 2]
```diff
@@ -329,9 +330,20 @@ const getFirst = (first, filter, user, recursive, unrestricted) => {
...
   // querying a single document by path is always allowed
   const path = getFilterValue(filter, 'path')
+
   const repoId = getFilterValue(filter, 'repoId')
   const oneRepoId = repoId && (!Array.isArray(repoId) || repoId.length === 1)
-  if (DOCUMENTS_RESTRICT_TO_ROLES && !recursive && !path && !oneRepoId && !unrestricted) {
+
+  const format = getFilterValue(filter, 'format')
+  const whitelisted = format && format.length && DOCUMENTS_UNRESTRICTED_REPO_IDS && DOCUMENTS_UNRESTRICTED_REPO_IDS
+    .split(',')
```
Comment (patrickvenetz) [inline packages/search/graphql/resolvers/_queries/search.js]: `DOCUMENTS_UNRESTRICTED_REPO_IDS` is split on each `getFirst` call, which is a bit over the top. Maybe prep variables for their use case on initialisation?

File: packages/search/graphql/resolvers/_queries/search.js [hunk 3]
```diff
@@ -324,14 +325,29 @@ const parseOptions = (options) => {
...
+
   const repoId = getFilterValue(filter, 'repoId')
   const oneRepoId = repoId && (!Array.isArray(repoId) || repoId.length === 1)
-  if (DOCUMENTS_RESTRICT_TO_ROLES && !recursive && !path && !oneRepoId && !unrestricted) {
+
+  const format = getFilterValue(filter, 'format')
+  const unrestricted =
+    format &&
+    format.length &&
+    DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS &&
```
Comment (tpreusse) [inline packages/search/graphql/resolvers/_queries/search.js]: This seems to check that *some* unrestricted repo id is present in format filter. Should't it check that **every** repo id in in the format filter is present in `DOCUMENTS_UNRESTRICTED_CHILDREN_REPO_IDS`?

Comment (patte) [reply packages/search/graphql/resolvers/_queries/search.js]: no. format is augmented with repoIds of other environments.  
```
format: [
  'https://github.com/republik/format-preis-der-republik',
  'https://github.com/republik-test/format-preis-der-republik',
  'https://github.com/republik-dev/format-preis-der-republik',
  'https://github.com/republik-dev/format-preis-der-republik'
]
```

Comment (tpreusse) [reply packages/search/graphql/resolvers/_queries/search.js]: Ah I see and multiple formats has it own key. So it's always one anyhow. Sorry didn't realize.

Comment (patte) [reply packages/search/graphql/resolvers/_queries/search.js]: no need to be sorry, it's absolutely not obvious.

**Pull request merged**